### ISSUE TITLE
Dahua debug console support for DVRs

### ISSUE THREAD

**Issue by speidy** [mcw0/Tools#3]
Hi @mcw0,

Is the Dahua debug console supported on Dahua DVR/NVRs? 

I couldn’t get it work there (tried both DVRIP and P2P)

Thanks

**Comment by mcw0** [comment]
Hi speidy,
Unfortunately not working to my best knowledge, the console is there and can be initiated but fails during attach, have not figured (yet) why it fails.

**Comment by arkanoid87** [comment]
I've just discovered your project (impressive!)
I'm also experiencing this on this hybrid nvr+camera (in local subnet) product.

https://www.dahuasecurity.com/asset/upload/product/20180905/MSB-L200.pdf

I'm also a python programmer, I hope to find some time to help on this

**Comment by arkanoid87** [comment]
you code has mixed tabs and spaces
my eyes are bleeding

**Comment by arkanoid87** [comment]
for my NVR (+camera), using --dump config was failing on configManager.getConfig (empty answer) so I changed to configManager.getConfigEx (I saw it in service dump) and I got config successfully

**Comment by mcw0** [comment]
> for my NVR (+camera), using --dump config was failing on configManager.getConfig (empty answer) so I changed to configManager.getConfigEx (I saw it in service dump) and I got config successfully

Yeah, known issue with my NVR too, the recv() loop timing out before anything been received, ugly fix if to increase the timeout manually with 5 sec or more. Not had time to try work out nice solution for that...

**Comment by mcw0** [comment]
Major rewrite coming up soon, tried to concentrate specially on NVR this time, still no success to attach to the console... looks like to me that the thread is locked and do not accept any attach.

However, instead of bailing out in the previous code, we will continue in the new code and start w/o attaching console.
Nevertheless, it do exist some quite useful functions in NVR's, implemented some of them in the new code. 

I think I'll commit the major rewrite in few weeks from now.
